ui/src/services/dataset.js

```diff
@@ -137,9 +145,50 @@ class DatasetService {
     return api.post(`/datasets/${dataset_id}/workflow/${workflow}`);
   }
 
+  check_if_exists({ name, type } = {}) {
+    return api.get(`/datasets/${type}/${name}/exists`);
+  }
+
   get_bundle_name(dataset) {
     return `${dataset.name}.${dataset.type}.tar`;
   }
+
```

Comment:
Below are the services to call the upload APIs which have now been made part of the `/datasets` route.